pkg/kademlia/workers.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (C) 2018 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package kademlia
+
+import (
+	"bytes"
+	"context"
+	"math/rand"
+	"sync"
+	"time"
+
+	"storj.io/storj/pkg/node"
+	proto "storj.io/storj/protos/overlay"
+)
+
+const (
+	uncontacted = iota
+	inProgress
+	completed
+)
+
+type chore struct {
+	status int
+	node   *proto.Node
+}
+
+type worker struct {
+	// node id to chore
+	contacted   map[string]*chore
+	mu          *sync.Mutex
+	maxResponse time.Duration
+	cancel      context.CancelFunc
+	nodeClient  node.Client
+	find        proto.Node
+	k           int
+}
+
+func (w *worker) work(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
```

Comment:
if this is all you're using the select for, you can just check if ctx.Err() is nil. If it's nil, nothing's been canceled.

Reply:
is that the preferred way? I've always seen it handled like this.